backend/inspirehep/records/marshmallow/utils.py

```diff
@@ -15,14 +15,15 @@ def get_display_name_for_author_name(author_name):
 
 
 def get_facet_author_name_for_author(author):
-    author_ids = author.get("ids", [])
-    author_bai = get_values_for_schema(author_ids, "INSPIRE BAI")
-    bai = author_bai[0] if author_bai else "BAI"
+    control_number = (
+        author["control_number"] if author.get("control_number") else "NOREC"
```

Comment:
There should always be a control number here. Maybe log an error if that's not the case and don't put it in the facet?

Reply:
but in the schema the control number isn't required, this is why I added this. Do you think this check should be skipped?
What do you mean by don't put it in the facet? Put NOREC instead? I am already doing this.

Reply:
Actually, I don't think that this check is necessary anymore - I think I was being paranoid, I'll remove it. This method is either called in the literature serializer (where we are already retrieving linked authors by id so it has to be there) or from the author details serializer.

Reply:
We had some records in the past without a control number. It shouldn't be happening, so I think it's OK to blow up.